lib/copy/__tests__/copy-symlink-same-target.test.js

```diff
@@ -0,0 +1,175 @@
+    it('should copy symlink when dest has no existing symlink', done => {
+      const target = path.join(TEST_DIR, 'target')
+      const src = path.join(TEST_DIR, 'src')
+      const dest = path.join(TEST_DIR, 'dest')
+
+      // Setup
+      fs.mkdirpSync(target)
+      fs.writeFileSync(path.join(target, 'file.txt'), 'content')
+      fs.mkdirpSync(src)
+      fs.symlinkSync(target, path.join(src, 'link'), 'dir')
+
+      // Copy
+      fs.copy(src, dest, err => {
+        assert.ifError(err)
+
+        // Verify symlink was copied
+        const destLink = path.join(dest, 'link')
+        assert.ok(fs.lstatSync(destLink).isSymbolicLink())
+        assert.strictEqual(fs.readlinkSync(destLink), target)
+        done()
+      })
+    })
+
+    it('should overwrite symlink when dest has existing symlink pointing to SAME target', done => {
+      const target = path.join(TEST_DIR, 'target')
+      const src = path.join(TEST_DIR, 'src')
+      const dest = path.join(TEST_DIR, 'dest')
+
+      // Setup
+      fs.mkdirpSync(target)
+      fs.writeFileSync(path.join(target, 'file.txt'), 'content')
+      fs.mkdirpSync(src)
+      fs.mkdirpSync(dest)
+      // Create symlinks pointing to the same target
+      fs.symlinkSync(target, path.join(src, 'link'), 'dir')
+      fs.symlinkSync(target, path.join(dest, 'link'), 'dir')
+
+      // Copy should work - two different symlinks pointing to same target are NOT the same file
+      fs.copy(src, dest)
+        .then(() => {
+          // Verify symlink was copied/overwritten
+          const destLink = path.join(dest, 'link')
+          assert.ok(fs.lstatSync(destLink).isSymbolicLink())
+          assert.strictEqual(fs.readlinkSync(destLink), target)
+          done()
+        })
+        .catch(done)
+    })
+
+    it('should overwrite symlink when dest has existing symlink pointing to DIFFERENT target', done => {
+      const target1 = path.join(TEST_DIR, 'target1')
+      const target2 = path.join(TEST_DIR, 'target2')
+      const src = path.join(TEST_DIR, 'src')
+      const dest = path.join(TEST_DIR, 'dest')
+
+      // Setup
+      fs.mkdirpSync(target1)
+      fs.mkdirpSync(target2)
+      fs.writeFileSync(path.join(target1, 'file.txt'), 'content1')
+      fs.writeFileSync(path.join(target2, 'file.txt'), 'content2')
+      fs.mkdirpSync(src)
+      fs.mkdirpSync(dest)
+      // Create symlinks pointing to different targets
+      fs.symlinkSync(target1, path.join(src, 'link'), 'dir')
+      fs.symlinkSync(target2, path.join(dest, 'link'), 'dir')
+
+      // Copy should work
+      fs.copy(src, dest, err => {
+        assert.ifError(err)
+
+        // Verify symlink was copied/overwritten to point to target1
+        const destLink = path.join(dest, 'link')
+        assert.ok(fs.lstatSync(destLink).isSymbolicLink())
+        assert.strictEqual(fs.readlinkSync(destLink), target1)
+        done()
+      })
+    })
+  })
+
+  describe('> when copying file symlinks', () => {
```

Comment:
Perhaps would be good to also have a `DIFFERENT target` test for file symlinks (as we do for directories)?